controller/space_template.go

```diff
@@ -0,0 +1,144 @@
+package controller
+
+import (
+	"encoding/base64"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/application"
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/jsonapi"
+	"github.com/almighty/almighty-core/login"
+	"github.com/almighty/almighty-core/rest"
+	"github.com/almighty/almighty-core/spacetemplate"
+	"github.com/goadesign/goa"
+)
+
+const APIStringType = "spacetemplates"
+
+// SpaceTemplateController implements the space_template resource.
+type SpaceTemplateController struct {
+	*goa.Controller
+	db     application.DB
+	config SpaceTemplateControllerConfiguration
+}
+
+// SpaceTemplateControllerConfiguration the configuration for the SpaceTemplateController
+type SpaceTemplateControllerConfiguration interface {
+	GetCacheControlSpaceTemplates() string
+}
+
+// NewSpaceTemplateController creates a space_template controller.
+func NewSpaceTemplateController(service *goa.Service, db application.DB, config SpaceTemplateControllerConfiguration) *SpaceTemplateController {
+	return &SpaceTemplateController{
+		Controller: service.NewController("SpaceTemplateController"),
+		db:         db,
+		config:     config,
+	}
+}
+
+// Create runs the create action.
+func (c *SpaceTemplateController) Create(ctx *app.CreateSpaceTemplateContext) error {
+	_, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+
+	newSpaceTemplate := spacetemplate.SpaceTemplate{
+		Name:        *ctx.Payload.Data.Attributes.Name,
+		Description: ctx.Payload.Data.Attributes.Description,
```

Comment:
Because a the `Description` field of `spacetemplate,SpaceTemplate` is a `string` **pointer**. It is optional to have a description on a space template.